strapi/strapi
# Fix TypeScript project builds with Generated Types

### What does it do?

- [DEVELOP MODE] Prevent typescript apps from crashing upon start or re-start when generated types are not up-to-date with the current context 
- [CLI] Allow generating types from the CLI even if the applications have compilation errors
- [NEW] Always generate application types on dev mode

### Why is it needed?

- Apps with generated types not matching the current state of their codebase cannot re-generate types because it currently asks for a successful compilation. The same thing happens on app restart (e.g. after creating a CT from the CTB) where newly created controllers & services don't match existing types (not generated yet)

### How to test it?

- Start an app in dev mode, and create a content type. The server should restart successfully & update the generated types accordingly
- Remove any reference to the newly created content type from the generated types, run `yarn strapi ts:generate-types`. It should throw an error and instead sync the types again

Edge cases:
- The app should still fail to start if there is another compilation error (not related to generated types)
- The app should successfully update types & start even if it's out of sync at startup (still dev mode only)

### Related issue(s)/PR(s)

fix #17084
fix #17906
fix #17962
